app/components/primer/beta/border_box.pcss

```diff
@@ -153,6 +143,18 @@
   }
 }
 
+/* Properly round corners if the first element of the Box is a row element */
+.Box > ul:first-child > li.Box-row:first-of-type {
```

Comment:
Unfortunately these styles are not being used for only Primer ViewComponents, but also for lots of custom markup on dotcom that use the `Box` classes directly. And sometimes it's a `<div>` or other HTML elements. E.g. [this example](https://github.com/github/github/blob/c42ea27d018b69e8704fa5e624c1dcc9671efb60/app/views/orgs/people/_sidebar.html.erb#L7-L8) would probably need:

```scss
.Box > .Box-row:first-of-type {
```

So I'm pretty sure this change would cause regressions where some `Box` use cases would miss the rounded corners due to custom markup use. 🤷 

Also see [this other comment](https://github.com/primer/view_components/issues/1937#issuecomment-1516124055).

Reply:
> We have implement a hack with hx to style it properly for our use case

Ok, great.

> or I can close this issue and leave it for your team to handle if thats more helpful at the moment.

Yeah, I think I'm gonna close this again until we can prioritize it. It just feels like it's something that needs extensive testing, just because there are hundreds of use cases. All with custom implementations from the time before we had ViewComponents and more control over the markup.

Thanks for the PR though.